lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1446,6 +1452,23 @@ ChangeResult TypeAnalyzer::visitAtenAddCLikeOp(
   return getLatticeElement(op->getResult(0)).join(knowledge);
 }
 
+ChangeResult TypeAnalyzer::visitAtenBitwiseAndTensorOp(
```

Comment:
I think this should be templated i.e., all the element-wise ops having type-promotion may need this template.

Reply:
This function is designed for the bitwise_and op, and I don't think it can be used for all the elementwise ops. Also, many elementwise ops are using different methods.